pkg/ipam/allocator/multipool/allocator.go

```diff
@@ -23,16 +26,29 @@ type Allocator struct {
 }
 
 func (a *Allocator) Init(ctx context.Context, logger *slog.Logger, _ *metrics.Registry) error {
-	a.poolAlloc = NewPoolAllocator(logger)
+	a.poolAlloc = NewPoolAllocator(logger, nil)
 	a.logger = logger.With(subsysLogAttr...)
 	return nil
 }
 
-func (a *Allocator) Start(ctx context.Context, getterUpdater ipam.CiliumNodeGetterUpdater, _ *metrics.Registry) (allocator.NodeEventHandler, error) {
+func (a *Allocator) Start(ctx context.Context, getterUpdater ipam.CiliumNodeGetterUpdater, reg *metrics.Registry) (allocator.NodeEventHandler, error) {
+	var iMetrics metricsAPI
+	if operatorOption.Config.EnableMetrics {
+		remainingIPsMetric := ipamMetrics.NewremainingIPsMetric(metrics.Namespace)
+		reg.Register(remainingIPsMetric.RemainingIPs)
+		iMetrics = remainingIPsMetric
+	} else {
+		a.logger.Info(" Metrics disabled")
+	}
+	a.poolAlloc.SetMetricsAPI(iMetrics)
 	return NewNodeHandler(a.logger, a.poolAlloc, getterUpdater), nil
 }
 
 func (a *Allocator) UpsertPool(ctx context.Context, pool *cilium_v2alpha1.CiliumPodIPPool) error {
+	if a.poolAlloc == nil {
+		return fmt.Errorf("allocator not initialized")
+	}
+
```

Comment:
This change seems to not be related to the purpose of the commit. Did you see a case in which `poolAlloc` was nil while calling `UpsertPool`? Looking at the current code it does not seem possible.